a2lfile/src/a2ml.rs

```diff
@@ -337,8 +409,9 @@ fn tokenize_a2ml(input: &str) -> Result<Vec<TokenType>, String> {
 // parse an a2ml fragment in an a2l file
 // The target data structure is the parsing definition used by the a2l parser, so that the
 // a2ml can control the parsing of IF_DATA blocks
-pub(crate) fn parse_a2ml(input: &str) -> Result<A2mlTypeSpec, String> {
-    let tok_result = tokenize_a2ml(input)?;
+pub(crate) fn parse_a2ml(filename: Option<String>, input: &str) -> Result<(A2mlTypeSpec, String), String> {
```

Comment:
like tokenize_a2ml: does the filename need to be an Option<>?

Reply:
done